src/main/java/nl/haarlem/translations/zdstozgw/controller/SoapController.java

```diff
@@ -49,42 +58,88 @@ public String beantwoordVraag(@RequestBody String body) {
         return response;
     }
 
-    @PostMapping(value = "/OntvangAsynchroon", consumes = MediaType.TEXT_XML_VALUE, produces = MediaType.TEXT_XML_VALUE)
-    public String ontvangAsynchroon(@RequestBody String body) {
+    private ZakLk01_v2 getZakLka01(String body) {
+        ZakLk01_v2 zakLk01 = null;
+        try {
+            zakLk01 = (ZakLk01_v2) JAXBContext.newInstance(ZakLk01_v2.class)
+                    .createUnmarshaller()
+                    .unmarshal(MessageFactory.newInstance().createMessage(null, new ByteArrayInputStream(body.getBytes())).getSOAPBody().extractContentAsDocument());
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        return zakLk01;
+    }
 
-        var stufRequest = new StufRequest(XmlUtils.convertStringToDocument(body));
+    private EdcLk01 getZakLEdcLk01(String body) {
+        EdcLk01 edcLk01 = null;
+        try {
+            edcLk01 = (EdcLk01) JAXBContext.newInstance(EdcLk01.class)
+                    .createUnmarshaller()
+                    .unmarshal(MessageFactory.newInstance().createMessage(null, new ByteArrayInputStream(body.getBytes())).getSOAPBody().extractContentAsDocument());
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        return edcLk01;
+    }
```

Comment:
Mooier zou 1 generieke functie zijn, de inhoud van de functies is praktisch hetzelfde. Je zou dan een Class object mee kunnen geven als parameter en een generiek Object als return type.